tests/utils/BackbeatClientMock.js

```diff
@@ -0,0 +1,67 @@
+'use strict'; // eslint-disable-line
+
+const assert = require('assert');
+const { errors } = require('arsenal');
+
+class BackbeatClientMock {
+    constructor(failures) {
+        this.failures = failures;
+
+        this.calls = {};
+        this.stubMethod('deleteObjectFromExpiration', {});
+    }
+
+    makeRetryableError() {
+        const err = errors.ServiceUnavailable.customizeDescription('failing on purpose');
+        err.retryable = true;
+        return err;
+    }
+
+    stubMethod(methodName, successResult, stubError) {
```

Comment:
I'm fine with it, but I feel the kind of processing done here could be taken care of by using [Sinon stubs](https://sinonjs.org/releases/latest/stubs/)